go.mod

```diff
@@ -1,32 +1,37 @@
 module go.temporal.io/sdk
 
-go 1.20
+go 1.21
+
+toolchain go1.21.1
```

Comment:
Curious, is this just an artifact of your tooling or was this change required?

Reply:
I just ran `go get ...` and `go mod tidy` but it may have been required due to the nexus sdk using `slog`.

Reply:
Just FYI we should remove this before merging to main, it seems to mess up out CI trying to test multiple versions of Go